src/dask_awkward/lib/io/parquet.py

```diff
@@ -33,7 +33,35 @@
 T = TypeVar("T")
 
 
-class _FromParquetFn(ColumnProjectionMixin):
+def report_failure(exception, *args, **kwargs):
+    return ak.Array(
+        [
+            {
+                "columns": [],
+                "args": [repr(a) for a in args],
+                "kwargs": [[k, repr(v)] for k, v in kwargs.items()],
+                "exception": type(exception).__name__,
+                "message": str(exception),
+            }
+        ]
+    )
+
+
+def report_success(columns, *args, **kwargs):
+    return ak.Array(
+        [
+            {
+                "columns": columns,
+                "args": [repr(a) for a in args],
+                "kwargs": [[k, repr(v)] for k, v in kwargs.items()],
+                "exception": None,
+                "message": None,
+            }
+        ]
+    )
```

Comment:
It may be useful at first in case we're somehow seeing different behavior from dak.necessary_columns vs. the final set of projected columns? Otherwise it seems like it might be heavy to have all those repeated columns.

Reply:
I suppose this is where having `bool | ReportSpec`, with a good default, type might be nice. I'd say this case with the columns is one where you want to be able to choose that functionality for the error report if you need it.

Reply:
Yea makes sense, it would mostly be useful for debugging